test/lan_items_helpers_test.rb

```diff
@@ -338,19 +338,35 @@ def check_GetItemUdev(key, expected_value)
   end
 end
 
-describe "LanItems#new_type_device" do
+context "When proposing device names candidates" do
   before(:each) do
     allow(Yast::LanItems).to receive(:find_type_ifaces).and_return([])
     allow(Yast::LanItems).to receive(:find_type_ifaces).with("eth").and_return(["eth0", "eth2", "eth3"])
   end
 
-  it "generates a valid device name" do
-    expect(Yast::LanItems.new_type_device("br")).to eql "br0"
-    expect(Yast::LanItems.new_type_device("eth")).to eql "eth1"
+  describe "LanItems#new_type_device" do
+    it "generates a valid device name" do
+      expect(Yast::LanItems.new_type_device("br")).to eql "br0"
+      expect(Yast::LanItems.new_type_device("eth")).to eql "eth1"
+    end
+
+    it "raises an error when no type is provided" do
```

Comment:
please test also empty type and also what happens with bogus type like `star_galaxy_network` ?

Reply:
nothing ... if type is only checked if empty or not. yast2-network doesn't allow manual type specification (I mean by user), so type parsing is on yast2 / NetworkInterfaces module